packages/eslint-plugin/docs/rules/only-throw-error.mdx

```diff
@@ -123,6 +123,13 @@ interface Options {
     | string
   )[];
 
+  /**
+   * Whether to allow rethrowing caught values that are not `Error` objects.
+   *
+   * @default false
+   */
+  allowRethrowing?: boolean;
```

Comment:
[Docs] The options mentions in this page don't explain why any of the options exist. I suppose that's existing/separate docs work for the other ones. But do you think it'd be reasonable to at least mention `allowRethrowing` in this PR?

If you'd rather them all tackled together as a followup, I wouldn't block.